app/models/proposal.rb

```diff
@@ -107,18 +108,46 @@ def remove_approver(email)
     approval.destroy
   end
 
-  def initialize_approvals()
-    if self.linear? && self.approvals.any?
-      self.approvals.update_all(status: 'pending')
-      self.approvals.first.make_actionable!
-    elsif self.parallel?
-      self.approvals.update_all(status: 'actionable')
+  # Set the approver list, from any start state
+  def approvers=(approver_list)
+    approvals = approver_list.each_with_index.map do |approver, idx|
+      approval = self.approval_for(approver)
+      approval ||= Approval.new(user: approver, proposal: self)
+      approval.position = idx + 1   # start with 1
+      approval
+    end
+    self.approvals = approvals
+    self.kickstart_approvals()
+    self.reset_status()
+  end
+
+  # Trigger the appropriate approval, from any start state
+  def kickstart_approvals()
+    actionable = self.approvals.actionable
+    pending = self.approvals.pending
+    if self.parallel?
+      pending.update_all(status: 'actionable')
+    elsif self.linear? && actionable.empty? && pending.any?
+      pending.first.make_actionable!
+    end
+    # otherwise, approvals are correct
+  end
+
+  def reset_status()
+    unless self.cancelled?   # no escape from cancelled
+      if self.all_approved?
+        self.update(status: 'approved')
+      else
+        self.update(status: 'pending')
+      end
     end
   end
```

Comment:
I don't _think_ so, but glad to have the second review. Emergency requests wouldn't have any approvers, so `all_approved?` would be triggered (unless the request was cancelled), keeping this in the `approved` status.